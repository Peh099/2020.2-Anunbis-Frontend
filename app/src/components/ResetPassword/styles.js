import styled from 'styled-components';
import Button from '../Button';

export const Container = styled.div`
  background-color: var(--lightWhite);
  position: absolute;
  display: flex;
  flex-direction: ${(props) => (props.direction ? props.direction : 'column')};
  width: min(55vw, 300px);
  height: min(25vh, 185px);
  margin-top: 60px;
  border-radius: 10px;
  font-family: 'Comfortaa', cursive;
  Input {
    width: min(50vw, 280px);
    align-self: flex-start;
  }
  @media (max-height: 330px) {
    display: none;
  }
`;

export const Header = styled.header`
  width: inherit;
  display: flex;
  height: 30px;
  background-color: var(--black);
  color: white;
  justify-content: center;
  align-items: center;
  margin-bottom: 20px;
  @media (max-height: 660px) {
    font-size: 10px;
  }
  div {
    display: flex;
    height: inherit;
    align-items: center;
    margin-left: auto;
    margin-right: auto;
    font-family: 'Comfortaa', cursive;
  }

  Button {
    margin-top: 0px;
    border-radius: 0;
    border: none;
    background-color: var(--lightRed);
    color: var(--white);
    height: inherit;
    margin-inline-start: 0px;
    margin-inline-end: inherit;

    &:hover,
    &focus {
      color: var(--black);
    }
  }
`;

export const Btn = styled(Button)`
  margin-top: 10px;
  margin-bottom: 15px;
  color: white;
  height: 30px;
  border: none;

  &:focus,
  &:hover {
    background-color: var(--darkBlue);
  }
`;

export const FeedBack = styled.div`
  display: flex;
  align-self: center;
  height: fit-content;
  width: fit-content;
  color: red;
`;

export const MsgLoading = styled.div`
  color: black;
  width: 100%;
  display: flex;
  align-self: center;
  justify-content: center;
`;