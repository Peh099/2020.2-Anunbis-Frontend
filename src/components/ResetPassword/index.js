import React from 'react';
import styled from 'styled-components';
import Button from '../Button';
import Form from '../Form';
import Input from '../Input';
import { yupResolver } from '@hookform/resolvers/yup';
import schema from './validations';
import { useForm } from "react-hook-form";

export default function ResetPassword({onClick, student}){

    const { register, handleSubmit, errors } = useForm({
        resolver: yupResolver(schema),
      });

    function onSubmit (data){

        const body = {
            oldPassword: data.old_password,
            newPassword: data.new_password,
            confirmNewPassword: data.confirm_new_password
        }

          console.log(data);
    }

    return(
        <Container>
            <Header>
                REDEFINIÇÃO DE SENHA
                <Button text='X' backColor='red' padding='2px 2px' onClick={onClick} />
            </Header>

            <Form onSubmit={handleSubmit(onSubmit)}>
                <Form.Field errorMsg={errors.old_password?.message} >
                    <Input type="password" text="Digite a senha atual" name="old_password" register={register}/>
                </Form.Field>

                <Form.Field errorMsg={errors.new_password?.message} >
                    <Input type="password" text="Digite a nova senha" name="new_password" register={register}/>
                </Form.Field>

                <Form.Field errorMsg={errors.confirm_new_password?.message} >
                    <Input type="password" text="Repita a nova senha" name="confirm_new_password" register={register}/>
                </Form.Field>

                <BtnConfirm type='submit' text='CONFIRMAR' backColor='#26A69A' padding='3px 3px' />
            </Form>

        </Container>
    )
}

const Container = styled.div`
    background-color: #FFFDE7;
    position: absolute;
    display: flex;
    flex-direction: ${props => props.direction?props.direction:'column'};
    width: 300px;
    height: fit-content;
    margin-top: 60px;
    border-radius: 10px;

    Input{
        width: 280px;
        margin-top: 20px;
        align-self: flex-start;
    };

`

const Header = styled.header`
    width: inherit;
    height: 30px;
    background-color: #212121;
    color: white;
    text-align: center;
    justify-content: center;
    display: flex;
    /* flex-direction: row; */
    align-items: center;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;

    Button{
        margin-top: 0px;
        border-radius: 0;
        margin-left: auto;
    };
`

const BtnConfirm = styled(Button)`
    margin-top: 15px;
    margin-bottom: 15px;
    color: white;
`