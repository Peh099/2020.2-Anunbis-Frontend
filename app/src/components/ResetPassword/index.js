import React from 'react';
import { yupResolver } from '@hookform/resolvers/yup';
import { useForm } from 'react-hook-form';
import Form from '../Form';
import Input from '../Input';
import schema from './validations';
import { Container, Header, Btn, FeedBack } from './styles';
import api from '../../services/Api';
import Users from '../../services/Users';

export default function ResetPassword({ onClick }) {
  const [feedBack, setFeedBack] = React.useState('');

  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  function makeFeedback() {
    return (
      <FeedBack>
        <p>Senha alterada com Sucesso</p>
      </FeedBack>
    );
  }

  function onSubmit(data) {
    console.log(data);
    const body = {
      password: data.new_password,
    };
    api
      .put(Users.whoAuthenticated().localStorageName, body)
      .then(() => {
        setFeedBack(makeFeedback());
      })
      .catch((error) => {
        console.log(error);
      });
  }

  return (
    <Container>
      <Header>
        <div>REDEFINIÇÃO DE SENHA</div>
      </Header>

      <Form onSubmit={handleSubmit(onSubmit)}>
        <Form.Field errorMsg={errors.new_password?.message}>
          <Input
            type="password"
            text="Digite a nova senha"
            name="new_password"
            register={register}
          />
        </Form.Field>
        <Form.Field errorMsg={errors.confirm_new_password?.message}>
          <Input
            type="password"
            text="Repita a nova senha"
            name="confirm_new_password"
            register={register}
          />
        </Form.Field>
        {feedBack}
        <Form.Footer>
          <Btn
            text="CANCELAR"
            backColor="#26A69A"
            padding="3px 3px"
            onClick={() => onClick()}
          />
          <Btn
            type="submit"
            text="ALTERAR"
            backColor="#26A69A"
            padding="3px 6px"
          />
        </Form.Footer>
      </Form>
    </Container>
  );
}
