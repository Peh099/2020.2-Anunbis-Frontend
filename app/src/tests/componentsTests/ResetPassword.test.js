import renderer from 'react-test-renderer';
import ResetPassword from '../../components/ResetPassword';

describe('Snapshot ResetPassword component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<ResetPassword />).toJSON()
    
    expect(tree).toMatchSnapshot()
    })
})