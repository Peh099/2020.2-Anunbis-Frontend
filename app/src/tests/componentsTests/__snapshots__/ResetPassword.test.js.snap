// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ResetPassword component matches the snapshot 1`] = `
<div
  className="sc-gKAaRy HhKjm"
>
  <header
    className="sc-iCoGMd kttjhk"
  >
    <div>
      REDEFINIÇÃO DE SENHA
    </div>
  </header>
  <form
    className="sc-dlnjwi kCxWFn"
    onSubmit={[Function]}
  >
    <div
      className="sc-bdnxRM bvZCUM"
    >
      <input
        className="sc-eCApnc doXpOj"
        name="new_password"
        placeholder="Digite a nova senha"
        type="password"
      />
      <div
        className="sc-gtsrHT jSglii error"
      />
    </div>
    <div
      className="sc-bdnxRM bvZCUM"
    >
      <input
        className="sc-eCApnc doXpOj"
        name="confirm_new_password"
        placeholder="Repita a nova senha"
        type="password"
      />
      <div
        className="sc-gtsrHT jSglii error"
      />
    </div>
    
    <div
      className="sc-hKFxyN eGOBqi"
    >
      <button
        className="sc-jSFjdj czcTCj sc-fujyAs kQMtOh"
        onClick={[Function]}
      >
        CANCELAR
      </button>
      <button
        className="sc-jSFjdj gDzklW sc-fujyAs kQMtOh"
        type="submit"
      >
        ALTERAR
      </button>
    </div>
  </form>
</div>
`;
